api_test.go

```diff
@@ -208,3 +211,52 @@ func TestHttpApi(t *testing.T) {
 
 	tests.TestApi(newNodeProvider(ctx))(t)
 }
+
+func Test_NewURLApiWithClient(t *testing.T) {
+	t.Skip()
+	var (
+		url                 = "127.0.0.1:65501"
+		headerToTest        = "Test-Header"
+		expectedHeaderValue = "thisisaheadertest"
+	)
+	server, err := testHTTPServer(url, headerToTest, expectedHeaderValue)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer server.Close()
+	go func() {
+		server.ListenAndServe()
+	}()
+	time.Sleep(time.Second * 2)
```

Comment:
We could avoid this sleep by doing:
```
l, err := net.Listen("tcp", url)
if err != nil {
    log.Fatal(err)
}

go server.Serve(l)
```

Reply:
Is this needed with `httpest`?

Reply:
Nope! Before `httptest.NewServer` returns, it starts the test server. All that's needed is for the caller to call `Close`.